### PR TITLE
[Issue 1367]: Report a warning on empty diff.namespace

### LINKED ISSUE #1367: print a warning when one or both namespaces to be diffed are Branch.empty
```
.temp> diff.namespace temp.base temp.merged

  The namespaces are identical.

.temp> diff.namespace temp.haed temp.merged

  The namespaces are identical.
```
🤔  I didn't notice I was adding "temp" but was already in .temp

Maybe something like
> The namespaces .temp.temp.base and .temp.temp.merged are empty.  Was there a typo?

or

> ⚠️ The namespace .temp.temp.haed is empty, was there a typo?
>
> Added: ...

### PR DESCRIPTION
**Pull request by shawn-bachlet**
## Overview
This PR is a targeted to fix #1367. The changes are small and localized. All tests are passing locally and the warnings printed exactly match those listed in the issue. 

## Implementation notes

This PR checks if either branch in a diff is empty and reports a warning if either (or both) are empty.

## Interesting/controversial decisions

I'm not crazy about the type of the constructor. I think there may be a more semantically clear type but, I couldn't think of it.

## Test coverage

I tested this manually by running the commands listed in the PR as well as making sure non-empty diffs still succeeded as well.

## Loose ends

The issue list a possible further improvement. This seemed like an additional improvement possibly beyond the scope of this issue? 
> Maybe some heuristic too for rejecting pushes with unrelated* histories. *Our histories are always related by the empty Branch, but maybe they wouldn't have to be?

### CODE REVIEW

**Comment by shawn-bachlet** [comment]
@mitchellwrosen it's here 😄

**Review by mitchellwrosen — APPROVED** [review]
Nice one! Looks great. Please add your name to `CONTRIBUTORS.markdown` at your leisure, either in this PR or after, and merge when ready.

**Comment by mitchellwrosen** [comment]
(Are you able to click the merge button?)

**Comment by mitchellwrosen** [comment]
Oops, I don't think CI ran until I clicked a button just now. There may be some transcript failures due to changed output, if diffing an empty namespace was under test. If so, here's how to fix locally:

```
stack build
stack exec transcripts
# if everything looks good
git add .
```

File: parser-typechecker/src/Unison/Codebase/Editor/Output.hs
Comment (mitchellwrosen) [inline parser-typechecker/src/Unison/Codebase/Editor/Output.hs:332]: Oops... we might want to consider this a failure instead (`True`). I'm not totally sure what the implications of that are.

**Pull request merged**
